query-node/mappings/src/content/channel.ts

```diff
@@ -544,3 +556,66 @@ export async function content_ChannelFundsWithdrawn({ store, event }: EventConte
 
   await store.save<ChannelFundsWithdrawnEvent>(rewardClaimedEvent)
 }
+
+export async function processChannelPaymentFromMember(
+  store: DatabaseManager,
+  event: SubstrateEvent,
+  memberId: MemberId,
+  message: IMakeChannelPayment,
+  [payeeAccount, amount]: [AccountId32, Balance]
+): Promise<ChannelPaymentMadeEvent> {
+  const member = await getMemberById(store, memberId)
+
+  // Only channel reward accounts are being checked right now as payment destination.
+  // Transfers to any other destination will be ignored by the query node.
+  const channel = await store.get(Channel, { where: { rewardAccount: payeeAccount.toString() } })
+  if (!channel) {
+    inconsistentState('Payment made to unknown channel reward account')
+  }
+
+  // Get payment context from the metadata
+  const getPaymentContext = async (msg: IMakeChannelPayment) => {
+    if (msg.channelId) {
+      const paymentContext = new PaymentContextChannel()
+      if (msg.channelId.toString() !== channel.id) {
+        invalidMetadata(
```

Comment:
> Since msg.channelId doesn't really matter, as we always use the channel queried by reward account anyway, perhaps the user shouldn't even need to provide channelId?

Yeah, totally makes sense, will address it.